src/wh_client.c

```diff
@@ -119,66 +120,117 @@ int wh_Client_Cleanup(whClient* c)
     return 0;
 }
 
-int wh_Client_EchoRequest(whClient* c, uint16_t size, const void* data)
+int wh_Client_SendRequest(whClient* c,
+        uint16_t group, uint16_t action,
+        uint16_t data_size, const void* data)
 {
-    int rc = 0;
-    whMessageCommLenData msg;
     uint16_t req_id = 0;
-    uint16_t req_type = WOLFHSM_MESSAGE_TYPE_COMM_ECHO;
+    uint16_t kind = WH_MESSAGE_KIND(group, action);
+    int rc = 0;
+
+     rc = wh_CommClient_SendRequest(c->comm,
+                WH_COMM_MAGIC_NATIVE, kind, &req_id,
+                data_size, data);
+    if (rc == 0) {
+        c->last_req_kind = kind;
+        c->last_req_id = req_id;
+    }
+    return rc;
+}
+
+int wh_Client_RecvResponse(whClient *c,
```

Comment:
null check all pointer args (`data` is checked in `wh_CommClient_RecvResponse`)

Reply:
fixed